pkg/cloud/services/compute/instance.go

```diff
@@ -410,30 +411,40 @@ func (s *Service) getServerNetworks(networkParams []infrav1.NetworkParam) ([]inf
 	return nets, nil
 }
 
-// Helper function for getting image ID from name.
-func (s *Service) getImageID(imageName string) (string, error) {
-	if imageName == "" {
-		return "", nil
-	}
+// Helper function for getting image id from name.
+func (s *Service) getImageIDFromName(imageName string) (string, error) {
+	var opts images.ListOpts
+
+	opts.Name = imageName
 
-	opts := images.ListOpts{
-		Name: imageName,
-	}
 	allImages, err := s.computeService.ListImages(opts)
 	if err != nil {
 		return "", err
 	}
 
 	switch len(allImages) {
 	case 0:
-		return "", fmt.Errorf("no image with the name %s could be found", imageName)
+		return "", fmt.Errorf("no image with the Name %s could be found", imageName)
 	case 1:
 		return allImages[0].ID, nil
 	default:
+		// this should never happen
 		return "", fmt.Errorf("too many images with the name, %s, were found", imageName)
 	}
 }
 
+// Helper function for getting image ID from name or ID.
+func (s *Service) getImageID(imageUUID, imageName string) (string, error) {
+	if imageUUID != "" {
+		// we return imageUUID without check
+		return imageUUID, nil
+	} else if imageName != "" {
+		return s.getImageIDFromName(imageName)
+	}
```

Comment:
We can omit the `else if` here.
```suggestion
	} 
	if imageName != "" {
		return s.getImageIDFromName(imageName)
	}
```